### ISSUE TITLE
Pipeline generation script should not generate jobs that we haven't selected in the target configuration screen

### ISSUE THREAD

**Issue by karianna** [AdoptOpenJDK/openjdk-build#1328]
```
Processing DSL script pipelines/build/common/create_job_from_template.groovy
Warning: (create_job_from_template.groovy, line 51) concurrentBuild is deprecated
Existing items:
    GeneratedJob{name='build-scripts/jobs/jdk8u'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-aix-ppc64-openj9'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-linux-ppc64le-openj9'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-linux-s390x-openj9'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-linux-x64-openj9'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-linux-x64-openj9-linuxXL'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-windows-x64-openj9-windowsXL'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-windows-x86-32-openj9'}
Unreferenced items:
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-aix-ppc64-hotspot'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-linux-aarch64-hotspot'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-linux-arm-hotspot'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-linux-ppc64le-hotspot'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-linux-x64-hotspot'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-linux-x64-hotspot-jfr'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-mac-x64-hotspot'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-mac-x64-openj9'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-mac-x64-openj9-macosXL'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-solaris-sparcv9-hotspot'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-solaris-x64-hotspot'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-windows-x64-hotspot'}
    GeneratedJob{name='build-scripts/jobs/jdk8u/jdk8u-windows-x64-openj9'}
```

**Comment by M-Davies** [comment]
@karianna has there been any movement on this outside of this issue? I'm currently investigating implementing concurrent pipelines at Adopt and have a suspicion that this might be a blocker

**Comment by karianna** [comment]
> @karianna has there been any movement on this outside of this issue? I'm currently investigating implementing concurrent pipelines at Adopt and have a suspicion that this might be a blocker

@M-Davies - Not that I'm aware of.  I do know that @johnoliver is looking at revisiting the pipeline rcode after some core fixes in Jenkins.

**Comment by M-Davies** [comment]
From what I got from John, it seems that the pipelines were originally regenerated since it was difficult to create build configs for personal builds outside of the main pipelines. The configurations were regenerated each time as a failsafe in case someone accidentally put in an invalid config. It was thought that the configs wouldn't change very often so it wouldn't matter too much if a default configuration was put into the code. https://github.com/AdoptOpenJDK/openjdk-build/pull/1003

Since the way configurations are created has been changed (for the better) since then, it's probably not necessary to keep having a default config any longer. I'm currently looking into removing it and running a test pipeline to see if it will survive without a default configuration
